thundra/plugins/trace/trace_plugin.py

```diff
@@ -31,144 +34,185 @@ def before_invocation(self, data):
+        self.trace_data['tags']['aws.lambda.arn'] = getattr(context,
+                                                           constants.CONTEXT_INVOKED_FUNCTION_ARN, None)
+        self.trace_data['tags']['aws.lambda.memory.limit'] = getattr(context,
+                                                                    constants.CONTEXT_MEMORY_LIMIT_IN_MB,
+                                                                    None)
+        self.trace_data['tags']['aws.lambda.log_group_name'] = getattr(context,
+                                                                      constants.CONTEXT_LOG_GROUP_NAME,
+                                                                      None)
+        self.trace_data['tags']['aws.lambda.log_stream_name'] = getattr(context,
+                                                                       constants.CONTEXT_LOG_STREAM_NAME,
+                                                                       None)
         report_data = {
             'apiKey': reporter.api_key,
-            'type': 'AuditData',
-            'dataFormatVersion': constants.DATA_FORMAT_VERSION,
+            'type': 'Trace',
+            'dataModelVersion': constants.DATA_FORMAT_VERSION,
             'data': self.trace_data
         }
+        if 'error' in data:
+            error = data['error']
+            error_type = type(error)
+            #Adding tags
+            self.trace_data['tags']['error'] = True
+            self.trace_data['tags']['error.kind'] = error_type.__name__
+            self.trace_data['tags']['error.message'] = str(error)
+            if hasattr(error, 'code'):
+                self.trace_data['tags']['error.code'] = error.code
+            if hasattr(error, 'object'):
+                self.trace_data['tags']['error.object'] = error.object
+            if hasattr(error, 'stack'):
+                self.trace_data['tags']['error.stack'] = error.stack
         reporter.add_report(report_data)
+        reporter.add_report(self.span_data_list)
+
+        self.tracer.flush_finished_spans()
+        self.flush_current_span_data()
+
+
+    def flush_current_span_data(self):
+        self.span_data_list.clear()
 
+    def build_span(self, span, data):
+        close_time = span.start_time + span.duration
+        context = data['context']
+        function_name = getattr(context, constants.CONTEXT_FUNCTION_NAME, None)
+
+        span_data = {
+            'id': span.context.span_id,
+            'type': "Span",
+            'agentVersion': '',
+            'dataModelVersion': constants.DATA_FORMAT_VERSION,
+            'applicationId': utils.get_application_id(context),
+            'applicationDomainName': span.domain_name or '',
+            'applicationClassName': span.class_name or '',
+            'applicationName': function_name,
+            'applicationVersion': getattr(context, constants.CONTEXT_FUNCTION_VERSION, None),
+            'applicationStage': '',
```

Comment:
`applicationStage` should be taken from environment variable `thundra_application_stage`. See java https://github.com/thundra-io/thundra-agent-java/blob/redesign/thundra-agent-core/src/main/java/io/thundra/agent/core/app/impl/PropertyAwareApplicationInfoProvider.java#L42

Reply:
I will update java part as `thundra_agent_lambda_application_stage`